File: libraries/managed-exceptions-lib/src/python/managed_exceptions/argumemts/invalid_argument_exception.py
from http import HTTPStatus
from managed_exceptions.error_details import ErrorDetails
from managed_exceptions.managed_exception import ManagedException

class InvalidArgumentException(ManagedException):
    def __init__(self, message: str, diagnostic_details: dict[str, str] = {}):
        super().__init__(ErrorDetails(
            status_code=HTTPStatus.BAD_REQUEST,
            diagnostic_code="00400",
            diagnostic_details=diagnostic_details,
            message=message
        ))

File: services/Sphana/Libraries/Sphana.ManagedException/Arguments/InvalidArgumentException.cs
using System.Net;

namespace Sphana.ManagedException.Arguments;

/// <summary>
/// Thrown when an invalid argument is provided to an operation.
/// </summary>
public class InvalidArgumentException : ManagedException
{
    public InvalidArgumentException(string message, Dictionary<string, string>? diagnosticDetails = null)
        : base(HttpStatusCode.BadRequest, "INVALID_ARGUMENT", message, diagnosticDetails)
    {
    }
}